src/peer_manager.rs

```diff
@@ -389,8 +401,11 @@ impl PeerManager {
 
         let names = self.routing_table.expect_join_our_section(expected_name)?;
 
-        let mut rng = SeededRng::new();
-        let seed = rng.gen_iter().take(10).collect();
+#[cfg(not(feature = "use-mock-crust"))]
+        let seed = SeededRng::new().gen_iter().take(10).collect();
```

Comment:
I didn't spot this during the last review, but we shouldn't be using `SeededRng` outside of tests I believe.  Using it like this will result in the same `seed` being generated every single time `expect_join_our_section()` is called.  It will only be different once the process restarts.

I think for non-tests, we should just stick to the `rand` crate.